src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -264,6 +274,19 @@ begin
     simp [hd] }
 end
 
+@[simp]
+lemma homogeneous_component_add :
+  homogeneous_component n (φ + ψ) = homogeneous_component n φ + homogeneous_component n ψ :=
+by rw [homogeneous_component, linear_map.comp_apply, linear_map.comp_apply, linear_map.comp_apply,
+    linear_map.map_add, linear_map.map_add]
+
+@[simp]
+lemma homogeneous_component_C_mul (n : ℕ) (r : R) :
+  homogeneous_component n (C r * φ) = C r * homogeneous_component n φ :=
+by simp only [homogeneous_component, finsupp.restrict_dom_apply,
+  submodule.subtype_apply, function.comp_app, linear_map.coe_comp,
+  set.mem_set_of_eq, C_mul', finsupp.filter_smul]
```

Comment:
Doesn't this follow directly from something relating `C_mul` to `smul`, and then using `linear_map.map_smul`? Perhaps we should even just have a `linear_map.map_polynomial_C_mul` lemma for this situation.

Reply:
Yeah I don't know what the squeeze simp before was doing, presumably something fairly similar, but this is shorter at least